### PR TITLE
Bugfix SOM realization

### PR DESCRIPTION
**Pull request by k4rst3ns**
## :bird: Description of this PR :bird:

- This PR will correct the buggy version of the cellpool som realization, that did not track correctly the noncropland land transitions.

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
- [x] Self-review own code
- [x] Document changes 
  - [x] Add changes to `CHANGELOG.md`
  - [x] Where relevant, put In-code documentation comments
  - [x] Properly address updates in interfaces in the module documentations
  - [x] [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
    [x] Run test runs via `Rscript start.R --> "test runs"`

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  *0.37* mins
  - This PR's with cellpool :  *0.41* mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate

### CODE REVIEW

File: modules/10_land/landmatrix_dec18/presolve.gms [hunk 1]
```diff
@@ -10,25 +10,25 @@
 *' @code Some of the land use transitions are restricted:
 
 *' No planted forest on natveg areas
-v10_lu_transitions.fx(j,"primforest","forestry") = 0;
-*v10_lu_transitions.fx(j,"secdforest","forestry") = 0;
-*v10_lu_transitions.fx(j,"other","forestry") = 0;
+vm_lu_transitions.fx(j,"primforest","forestry") = 0;
+*vm_lu_transitions.fx(j,"secdforest","forestry") = 0;
```
Comment (tscheypidi) [inline modules/10_land/landmatrix_dec18/presolve.gms:14]: can these commented lines and the following ones be removed? I know they were there before, but in general we should not have commented lines of code in there

File: modules/10_land/module.gms [hunk 2]
```diff
@@ -15,6 +15,5 @@
 *' @authors Jan Philipp Dietrich, Florian Humpenoeder, Kristine Karstens
```
Comment (tscheypidi) [inline modules/10_land/module.gms:16]: the module description should now also mention that the transition information is shared as well.

**Review by tscheypidi — CHANGES_REQUESTED** [review]
looks good...just two minor requests

File: modules/10_land/landmatrix_dec18/presolve.gms (hunk 1 above)
Comment (flohump) [reply modules/10_land/landmatrix_dec18/presolve.gms:14]: removing these lines is fine with me

File: modules/42_water_demand/agr_sector_aug13/input.gms [hunk 3]
```diff
@@ -33,7 +33,7 @@ s42_env_flow_base_fraction         Fraction of available water that is reserved
 * 									(determined in the file EFR_protection_policy.csv)
 s42_env_flow_fraction              Fraction of available water that is reserved for under protection policies (1) / 0.2 /
 s42_pumping                        Switch to activate pumping cost settings (1) / 0 /
-s42_multiplier_startyear           Year from which pumping costs multiplier will be implemented (1) / 0 /
+s42_multiplier_startyear           Year from which pumping costs multiplier will be implemented (1) / 1995 /
```
Comment (flohump) [inline modules/42_water_demand/agr_sector_aug13/input.gms:36]: I thought that we had sorted out this problem @tscheypidi what do you think?

Comment (flohump) [reply modules/42_water_demand/agr_sector_aug13/input.gms:36]: 1995 is the correct value from the default.cfg. All fine here.

**Review by flohump — CHANGES_REQUESTED** [review]
Thanks for fixing this bug.
I have only some minor commetns , on top of @tscheypidi comments.

File: modules/10_land/landmatrix_dec18/presolve.gms (hunk 1 above)
Comment (k4rst3ns) [reply modules/10_land/landmatrix_dec18/presolve.gms]: removed all commented lines:

*v10_lu_transitions.fx(j,"secdforest","forestry") = 0;
*v10_lu_transitions.fx(j,"other","forestry") = 0;

*' Forestry can only increase
*v10_lu_transitions.fx(j,"forestry",land_to10) = 0;
*v10_lu_transitions.up(j,"forestry","forestry") = Inf;

*' Secdforest can only decrease (during optimization)
*v10_lu_transitions.fx(j,land_from10,"secdforest") = 0;
*v10_lu_transitions.up(j,"secdforest","secdforest") = Inf;

File: modules/10_land/module.gms (hunk 2 above)
Comment (k4rst3ns) [reply modules/10_land/module.gms:16]: the realization description has this info. Is it needed to move the information also here?

and speaking of the realization.gms, should I add the only net LU transitions per cluster issue to the limitations there?

File: modules/10_land/landmatrix_dec18/equations.gms [hunk 4]
```diff
@@ -17,41 +17,30 @@
...
-
-*' Additionally the following two equations calculate the sources and targets
-*' of all cropland related conversions.
-
-q10_croplandreduction(j2,land_to10) ..
-               vm_croplandreduction(j2,land_to10) =e=
-               v10_lu_transitions(j2,"crop",land_to10);
-
-q10_croplandexpansion(j2,land_from10) ..
-               vm_croplandexpansion(j2,land_from10) =e=
```
[the remark this answers is not in the document]
Comment (k4rst3ns) [reply modules/10_land/landmatrix_dec18/equations.gms:53]: fixed. Also added that vm_lu_transitions is now an interface.

File: modules/10_land/module.gms (hunk 2 above)
Comment (tscheypidi) [reply modules/10_land/module.gms:16]: interfaces must be documented in module.gms are valid for all realizations. Adding the warning that it is only net transitions to limitations is a good idea.

**Review by tscheypidi — APPROVED** [review]

**Review by flohump — APPROVED** [review]
Looks good.

**Pull request merged**